REVEL.pm

```diff
@@ -108,17 +114,36 @@ sub run {
 sub parse_data {
   my ($self, $line) = @_;
 
-  my ($c, $s, $ref, $alt, $refaa, $altaa, $revel_value) = split /\t/, $line;
-
-  return {
-    alt => $alt,
-    start => $s,
-    end => $s,
-    altaa => $altaa,
-    result => {
-      REVEL   => $revel_value,
-    }
-  };
+  my @values = split /\t/, $line;
+  # the first version only had GRCh37 coordinates
+  if (scalar @values == 7) {
```

Comment:
For a given REVEL input file the processing should be the same. The number of fields is being checked for each line returned. Could this file be checked earlier to see if contained 7 or 8 columns and then this used to process the lines? For example dbNSFP plugin checks format in sub run. If the last column is empty, the split will ignore trailing empty lines and could result in lines being processed differently.

Reply:
Thank you for the suggestions. I get the column number now from the header in the new sub.